gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigResourceLocalHandler.java

```diff
@@ -205,9 +205,19 @@ public static FlowSpec createFlowSpecForConfig(FlowConfig flowConfig) {
     }
 
     Config config = configBuilder.build();
-    //We allow for flowConfig to be in HOCON format. We first convert the StringMap object to a String object and
-    // then use ConfigFactory#parseString() to parse the HOCON string.
-    Config configWithFallback = config.withFallback(ConfigFactory.parseString(flowConfig.getProperties().toString()).resolve());
+
+    Config configWithFallback;
+    //We first attempt to process the REST.li request as a HOCON string. If the request is not a valid HOCON string
+    // (e.g. when certain special characters such as ":" or "*" are not properly escaped), we catch the Typesafe ConfigException and
+    // fallback to assuming that values are literal strings.
+    try {
+      // We first convert the StringMap object to a String object and then use ConfigFactory#parseString() to parse the
+      // HOCON string.
+      configWithFallback = config.withFallback(ConfigFactory.parseString(flowConfig.getProperties().toString()).resolve());
+    } catch (Exception e) {
+      configWithFallback = config.withFallback(ConfigFactory.parseMap(flowConfig.getProperties()));
```

Comment:
Thanks for the comment! There is a problem with making it config driven. We do not want to make it a service-wide config i.e. all requests must conform to one or the other. The config cannot be part of the request, since the config will determine how to parse the request. Hence, I went for the try-catch approach.

Reply:
@sv2000 I'm okay with the implementation. But it would be much safer if we could add test cases to understand the difference between these two implementations. In LinkedIn internal datapacks, people would have double quotes escaped on purpose (please refer to https://jarvis.corp.linkedin.com/codesearch/result/?name=lss_dynamics_contact_201708161641_PROD.template&path=gobblin-service-datapacks%2Fgobblin-service-datapacks%2Fsrc%2Fmain%2Fresources%2Ftemplates%2Fprod&reponame=etl-infra-tools%2Fgobblin-service-datapacks). I'm not sure if those implementation would be affected by changing it to parseString() API. Bad situation could be the parseString() would return successfully but end up with different results.